tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause.bal

```diff
@@ -74,6 +78,63 @@ function testMultiLevelOnFail() {
     assertEquality(" -> Before error thrown,  -> Error caught at level #1 -> Before error thrown,  -> Error caught at level #1", str);
 }
 
+function testMultiLevelOnFailWithoutVariableVariationOne() {
+    int i = 0;
+    string str = "";
+
+    while i < 2 {
+        do {
+            str += " -> B, ";
+            fail getError();
+        } on fail {
+            str += " -> Error #1";
+        }
+        i = i + 1;
+    } on fail error e {
+        str += " -> Error #2";
+    }
+
+    assertEquality(" -> B,  -> Error #1 -> B,  -> Error #1", str);
```

Comment:
Again, not the easiest test to read with the arbitrary strings. IMO, even just letters or numbers will be easier to follow.